### DISCUSSION TITLE
Real time: collection name and record id

### DISCUSSION THREAD

**Discussion by ilteoood** [pocketbase/pocketbase#3255] [Q&A] [answered]
I'm trying to use the `onRealtimeConnectRequest` and `onRealtimeDisconnectRequest` hooks in order to increase an internal counter when an user connects/disconnects.
Unfortunately, I can't find in any of them the collection name and record id that the user is subscribing to.

Do you have an example about how to achieve that?

**Comment by ganigeorgiev** [comment, marked as answer]
`onRealtimeConnectRequest` hook is triggered only once, on the initial client connect request.

The subscriptions are submitted **after** the connection was established (if at all).

I'm not sure that I understand your use case but you can try to listen to the [`OnRealtimeAfterSubscribeRequest` hook](https://pocketbase.io/docs/js-event-hooks/#onrealtimeaftersubscriberequest) to track subscriptions changes.

Note that the above hook is not triggered on disconnect so depending on your use case you may also have to listen to the [`onRealtimeDisconnectRequest` hook](https://pocketbase.io/docs/js-event-hooks/#onrealtimedisconnectrequest) to cleanup/reset the counter.

You should be able to also access all currently active realtime clients and their subscriptions using something like [`$app.subscriptionsBroker().clients()[clientId].subscriptions()`](https://pocketbase.io/jsvm/interfaces/subscriptions.Broker.html).
